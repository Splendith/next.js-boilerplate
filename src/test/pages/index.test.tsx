import React from 'react';
import { render } from '../testUtils';

// @ts-ignore
import preloadAll from 'jest-next-dynamic';

import Index from '../../pages/index';

beforeAll(async () => {
  await preloadAll();
});

describe('Index', () => {
  const elem = <Index />;
  it('should shows "Welcome to this website!"', async () => {
    const app = render(elem, {});
    expect(app.getByText('Welcome to this website!')).toBeInTheDocument();
  });
  it('matches snapshot', () => {
    const { asFragment } = render(elem, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
