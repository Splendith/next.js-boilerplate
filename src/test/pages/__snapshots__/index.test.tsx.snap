// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Wrapper-gu0x4c-0 kryztR"
  >
    <div
      class="Box-gu0x4c-1 eWkUHl"
    >
      <h1
        class="Title-fubljw-0 bFihJi"
      >
        Hello world!
      </h1>
      <div
        class="Image-sc-1ooavon-0 cqWqZz"
      >
        <img
          alt="Smile"
          src="/static/img/smile.png"
        />
      </div>
      <p
        class="Welcome-sc-1ooavon-1 eNuoct"
      >
        Welcome to this website!
      </p>
      <a
        href="/testurl/5?queryId=8"
      >
        route=“/test/5?queryId=8“
      </a>
    </div>
    <div
      class="Box-sc-1hlek3p-0 iofyeJ"
    >
      <h5>
        Recoil global state
      </h5>
      <button
        class="btn btn-primary"
        type="button"
      >
        Add count
      </button>
       
      <button
        class="btn btn-outline-secondary"
        type="button"
      >
        Clear count
      </button>
      <br />
      <br />
      <div>
        Counter: 0
      </div>
    </div>
  </div>
</DocumentFragment>
`;
